src/components/organisms/EntityList/EntityListFilters/LabelsFilter/LabelsFilter.tsx

```diff
@@ -65,35 +75,54 @@ const LabelsFilter: React.FC<FilterProps> = props => {
     setLabelsMapping(mapping.length === 0 ? [defaultKeyValuePair] : mapping);
   }, [filters.selector]);
 
-  const renderKeyValueInputs = labelsMapping.map((item, index) => {
-    const key = `key-value-pair${index}`;
-
-    return (
-      <StyledKeyValueRow key={key}>
-        <Input
-          width="220px"
-          onChange={event => onKeyChange(event.target.value, index)}
-          value={item.key}
-          data-cy={`key-input-${index}`}
-          placeholder="Key"
-        />
-        <Input
-          width="220px"
-          onChange={event => onValueChange(event.target.value, index)}
-          value={item.value}
-          data-cy={`value-input-${index}`}
-          placeholder="Value"
-        />
-        {index > 0 ? (
-          <Button $customType="tertiary" onClick={() => onDeleteRow(index)} data-cy={`delete-row-${index}`}>
-            &#10005;
-          </Button>
-        ) : (
-          <EmptyButton />
-        )}
-      </StyledKeyValueRow>
-    );
-  });
+  const keysLabels = useMemo(() => Object.keys(data ?? {}).map(key => ({key, label: key, value: key})), [data]);
+
+  const ValuesLabels = useMemo(
+    () => (data ? keysLabels.map(item => data[item.key].map(v => ({key: item.key, value: v}))).flat() : []),
+    [keysLabels]
+  );
+
+  const renderKeyValueInputs = useMemo(
+    () =>
+      labelsMapping.map((item, index) => {
+        const key = `key-value-pair${index}`;
+
+        const keyOptions = keysLabels.filter(f => f.key.startsWith(item.key));
+
+        const valuesOptions = ValuesLabels.filter(
+          f => f.key.startsWith(item.key) && f.value.startsWith(item.value)
```

Comment:
Probably:
```suggestion
          f => f.key === item.key && f.value.startsWith(item.value)
```

Reply:
`startsWith` to show all possible suggestions.

Reply:
But these are options for the value.
* If the `key = "abc"`, should it autocomplete values also from `key = "abcdef"`? (that's what `f.key.startsWith()` will mean)
* when it comes to `f.value.startsWith` - I initially thought that it will filter it automatically, but I checked and `startsWith` is needed here 👍

Reply:
Looks like it's still not resolved

Reply:
I'll resolve this comment and create the same new one, to avoid hiding it 👍